File: book/python爬虫/多进程与多进程爬虫/thread.java
请把以下Java代码转换成Python代码
public class thread {
    public static void main(String[] args) throws InterruptedException {
        System.out.println("-----主线程开始-----");

        List<Thread> threads = new ArrayList<>();
        for (int i = 0; i < 4; i++) {
            Thread t = new Thread(() -> {
                for (int j = 0; j < 3; j++) {
                    try {
                        Thread.sleep(1000);
                        System.out.printf("Thread's name is %s\n", Thread.currentThread().getName());
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            });
            threads.add(t);
        }

        for (Thread t : threads) {
            t.start();
        }

        for (Thread t : threads) {
            t.join();
        }

        System.out.println("-----主线程结束-----");
    }
}

File: book/python爬虫/多进程与多进程爬虫/thread.py
from threading import Thread, Lock
import time

n = 100

def task():
    global n
    mutex.acquire()
    temp = n
    time.sleep(0.1)
    n = temp - 1
    print(f'购买成功，剩余{n}张电影票')
    mutex.release()

if __name__ == "__main__":
    mutex= Lock()
    t_l = []
    for i in range(10):
        t=Thread(target=task)
        t_l.append(t)
        t.start()
    for t in t_l:
        t.join()